django/db/models/fields/related_descriptors.py

```diff
@@ -697,6 +697,10 @@ def remove(self, *objs, bulk=True):
                 val = self.field.get_foreign_related_value(self.instance)
                 old_ids = set()
                 for obj in objs:
+                    if not isinstance(obj, self.model):
+                        raise TypeError("'%s' instance expected, got %r" % (
+                            self.model._meta.object_name, obj,
+                        ))
```

Comment:
Shouldn't this new error add in a separate commit?

Reply:
So currently there are two commits: one for the docs change. One for clarifying the behaviour (which adds this error and the tests.) 

🤷‍♂